src/api/telemetry/TelemetryAPI.js

```diff
@@ -250,6 +250,121 @@ export default class TelemetryAPI {
     return options;
   }
 
+  /**
+   * Canonicalizes options object by creating a standardized version with sorted keys
+   * and recursively processed values. This ensures consistent object representation
+   * regardless of property order.
+   *
+   * Used internally to generate consistent hashes for caching subscription options.
+   *
+   * @private
+   * @param {Object|Array|*} options The options object to canonicalize
+   * @returns {Object|Array|*} A canonicalized version of the input where object keys are sorted
+   */
+  #canonicalizeOptions(options) {
+    if (typeof options !== 'object' || options === null) {
+      return options;
+    }
+
+    if (Array.isArray(options)) {
+      return options.map((item) => this.#canonicalizeOptions(item));
+    }
+
+    return Object.keys(options)
+      .sort()
+      .reduce((acc, key) => {
+        acc[key] = this.#canonicalizeOptions(options[key]);
+        return acc;
+      }, {});
+  }
+
+  /**
+   * Filters out class instances and functions from an object, returning a plain
+   * data object suitable for JSON serialization.
+   *
+   * @private
+   * @param {Object|Array|*} value The value to sanitize
+   * @returns {Object|Array|*} A sanitized copy with class instances and functions removed
+   */
+  #sanitizeForSerialization(value) {
+    if (value === null || value === undefined) {
+      return value;
+    }
+
+    // Handle primitive types, arrays and objects
+    if (typeof value !== 'object') {
+      return typeof value !== 'function' ? value : undefined;
+    }
+
+    if (Array.isArray(value)) {
+      return value.map((item) => this.#sanitizeForSerialization(item));
+    }
+
+    if (Object.getPrototypeOf(value) === Object.prototype) {
+      const result = {};
+
+      for (const key of Object.keys(value)) {
+        const sanitizedValue = this.#sanitizeForSerialization(value[key]);
+
+        if (sanitizedValue !== undefined) {
+          result[key] = sanitizedValue;
+        }
+      }
+      return result;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Generates a numeric hash value for an options object. The hash is consistent
+   * for equivalent option objects regardless of property order.
+   *
+   * This is used to create compact, unique cache keys for telemetry subscriptions with
+   * different options configurations. The hash function ensures that identical options
+   * objects will always generate the same hash value, while different options objects
+   * (even with small differences) will generate different hash values.
+   *
+   * @private
+   * @param {Object} options The options object to hash
+   * @returns {number} A positive integer hash of the options object
+   */
+  #hashOptions(options) {
+    const sanitizedOptions = this.#sanitizeForSerialization(options);
```

Comment:
This seems to be doing the same job as [replacer functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter) in JSON.stringify. We might be able to do this all in a single call to JSON.stringify and save some CPU cycles.

Reply:
I've modified sanitize for serialization to be the replacer function!